### ISSUE TITLE
Webportal sometimes display wrong GPU usage ratio

### ISSUE THREAD

**Issue by ICEORY** [Microsoft/pai#1949]
OpenPAI environment:
- OpenPAI v0.8.2, Ubuntu16.04
- GPU: geforce1080ti

Issues: Webportal sometimes display wrong GPU usage ratio

It seems that the node exporter is unstable and sometimes failed to get the gpu information from worker nodes. 

![image](https://user-images.githubusercontent.com/11911903/50468729-49fcab80-09e4-11e9-82c8-abe94235348a.png)

- Screenshot -1 
![screenshot from 2018-12-27 14-07-43](https://user-images.githubusercontent.com/11911903/50468513-3270f300-09e3-11e9-99b0-106b8fe26c3b.png)
![screenshot from 2018-12-27 14-07-15](https://user-images.githubusercontent.com/11911903/50468511-2ab14e80-09e3-11e9-8431-51a446b3ace4.png)

- Screenshot -2
![screenshot from 2018-12-27 14-05-47](https://user-images.githubusercontent.com/11911903/50468532-58969300-09e3-11e9-89ce-9bf334250df5.png)
![screenshot from 2018-12-27 14-06-07](https://user-images.githubusercontent.com/11911903/50468533-59c7c000-09e3-11e9-9a8a-9fa110554262.png)

**Comment by ICEORY** [comment]
- Screenshot -3
![screenshot from 2018-12-27 14-38-21](https://user-images.githubusercontent.com/11911903/50468926-3aca2d80-09e5-11e9-9998-060488703a51.png)
![screenshot from 2018-12-27 14-38-37](https://user-images.githubusercontent.com/11911903/50468936-41f13b80-09e5-11e9-98ba-145d500696f5.png)

**Comment by ICEORY** [comment]
I copy the log from webportal

- job  exporter
```
2018-12-29 09:51:18,899 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:19,842 - DEBUG - job_exporter.py:107 - pid 14805 with cmd `/bin/bash /usr/local/start.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae410&gt;, {'24': set(['192.168.xxx.xxx:59522', '192.168.xxx.xxx:5010'])}), in 168076, out 22282132
2018-12-29 09:51:19,863 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:21,724 - DEBUG - job_exporter.py:107 - pid 22822 with cmd `/bin/sh -c ./install-all-drivers
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae8c0&gt;, {'2213': set(['192.168.xxx.xxx:22']), '18307': set(['192.168.xxx.xxx:22']), '28236': set(['192.168.xxx.xxx:55088', '192.168.xxx.xxx:33500', '192.168.xxx.xxx:47098', '192.168.xxx.xxx:56792', '192.168.xxx.xxx:51590', '192.168.xxx.xxx:55394', '192.168.xxx.xxx:34438']), '6012': set(['192.168.xxx.xxx:48680', '192.168.xxx.xxx:51476', '192.168.xxx.xxx:47000', '192.168.xxx.xxx:55006', '192.168.xxx.xxx:33536', '192.168.xxx.xxx:51388', '192.168.xxx.xxx:56784', '192.168.xxx.xxx:51494', '192.168.xxx.xxx:34356']), '23877': set(['192.168.xxx.xxx:50128']), '17978': set(['192.168.xxx.xxx:22']), '33090': set(['192.168.xxx.xxx:26134']), '14951': set(['192.168.xxx.xxx:59522', '192.168.xxx.xxx:5010']), '2129': set(['192.168.xxx.xxx:22']), '29758': set(['192.168.xxx.xxx:32882', '192.168.xxx.xxx:36648']), '20069': set(['192.168.xxx.xxx:30389']), '48563': set(['192.168.xxx.xxx:30389']), '24940': set(['192.168.xxx.xxx:41364', '192.168.xxx.xxx:40126', '192.168.xxx.xxx:36360', '192.168.xxx.xxx:53600']), '20637': set(['192.168.xxx.xxx:20052']), '22889': set(['192.168.xxx.xxx:9100']), '22215': set(['192.168.xxx.xxx:46892', '192.168.xxx.xxx:46896', '192.168.xxx.xxx:46894', '192.168.xxx.xxx:47104', '192.168.xxx.xxx:46898', '192.168.xxx.xxx:46900', '192.168.xxx.xxx:10250', '192.168.xxx.xxx:46928']), '4596': set(['192.168.xxx.xxx:13955']), '19788': set(['192.168.xxx.xxx:50916', '192.168.xxx.xxx:47220']), '15501': set(['192.168.xxx.xxx:25214']), '806': set(['192.168.xxx.xxx:34492', '192.168.xxx.xxx:33446', '192.168.xxx.xxx:51488', '192.168.xxx.xxx:55106', '192.168.xxx.xxx:56790', '192.168.xxx.xxx:51634', '192.168.xxx.xxx:47096']), '22424': set(['192.168.xxx.xxx:46902', '192.168.xxx.xxx:46910'])}), in 212328, out 22299421
2018-12-29 09:51:21,747 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:23,328 - DEBUG - job_exporter.py:107 - pid 18828 with cmd `/dev/init -- /bin/bash /pai/bootstrap/docker_bootstrap.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae6e0&gt;, {'5382': set(['192.168.xxx.xxx:25214'])}), in 26, out 88
2018-12-29 09:51:23,329 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:24,049 - DEBUG - job_exporter.py:107 - pid 362 with cmd `/dev/init -- /bin/bash /pai/bootstrap/docker_bootstrap.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae410&gt;, {'273': set(['192.168.xxx.xxx:13955']), '242': set(['192.168.xxx.xxx:55310', '192.168.xxx.xxx:47410', '192.168.xxx.xxx:51488', '192.168.xxx.xxx:57156', '192.168.xxx.xxx:33856', '192.168.xxx.xxx:52074', '192.168.xxx.xxx:34792'])}), in 0, out 0
2018-12-29 09:51:24,050 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:24,265 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:24,332 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:24,842 - DEBUG - job_exporter.py:107 - pid 22889 with cmd `/bin/node_exporter --collector.textfile.directory=/datastorage/prometheus --no-collector.arp --no-collector.bcache --no-collector.bonding --no-collector.conntrack --no-collector.edac --no-collector.entropy --no-collector.hwmon --no-collector.infiniband --no-collector.ipvs --no-collector.mdadm --no-collector.nfs --no-collector.nfsd --no-collector.sockstat --no-collector.stat --no-collector.time --no-collector.timex --no-collector.vmstat --no-collector.wifi --no-collector.xfs --no-collector.zfs
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae668&gt;, {'2213': set(['192.168.xxx.xxx:22']), '18307': set(['192.168.xxx.xxx:22']), '28236': set(['192.168.xxx.xxx:52224', '192.168.xxx.xxx:35016', '192.168.xxx.xxx:55572', '192.168.xxx.xxx:47430', '192.168.xxx.xxx:52228', '192.168.xxx.xxx:52200', '192.168.xxx.xxx:34060', '192.168.xxx.xxx:55394']), '6012': set(['192.168.xxx.xxx:47626', '192.168.xxx.xxx:48680', '192.168.xxx.xxx:55564', '192.168.xxx.xxx:34970', '192.168.xxx.xxx:35020', '192.168.xxx.xxx:57302', '192.168.xxx.xxx:34042', '192.168.xxx.xxx:52136']), '23877': set(['192.168.xxx.xxx:50128']), '17978': set(['192.168.xxx.xxx:22']), '33090': set(['192.168.xxx.xxx:26134']), '14951': set(['192.168.xxx.xxx:59522', '192.168.xxx.xxx:5010']), '2129': set(['192.168.xxx.xxx:22']), '29758': set(['192.168.xxx.xxx:32882', '192.168.xxx.xxx:36648']), '20069': set(['192.168.xxx.xxx:30389']), '48563': set(['192.168.xxx.xxx:30389']), '24940': set(['192.168.xxx.xxx:41364', '192.168.xxx.xxx:40126', '192.168.xxx.xxx:36360', '192.168.xxx.xxx:53600']), '20637': set(['192.168.xxx.xxx:20052']), '22889': set(['192.168.xxx.xxx:9100']), '22215': set(['192.168.xxx.xxx:46892', '192.168.xxx.xxx:46896', '192.168.xxx.xxx:46894', '192.168.xxx.xxx:47104', '192.168.xxx.xxx:46898', '192.168.xxx.xxx:46900', '192.168.xxx.xxx:10250', '192.168.xxx.xxx:46928']), '4596': set(['192.168.xxx.xxx:13955']), '19788': set(['192.168.xxx.xxx:50916', '192.168.xxx.xxx:47220']), '15501': set(['192.168.xxx.xxx:25214']), '806': set(['192.168.xxx.xxx:57278', '192.168.xxx.xxx:55310', '192.168.xxx.xxx:47410', '192.168.xxx.xxx:51488', '192.168.xxx.xxx:52074', '192.168.xxx.xxx:33856', '192.168.xxx.xxx:57280', '192.168.xxx.xxx:55624', '192.168.xxx.xxx:34792', '192.168.xxx.xxx:55622']), '22424': set(['192.168.xxx.xxx:46902', '192.168.xxx.xxx:46910'])}), in 212328, out 22299421
2018-12-29 09:51:24,843 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:25,432 - DEBUG - job_exporter.py:107 - pid 27737 with cmd `/dev/init -- /bin/bash /pai/bootstrap/docker_bootstrap.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae6e0&gt;, {'268': set(['192.168.xxx.xxx:26134']), '237': set(['192.168.xxx.xxx:35016', '192.168.xxx.xxx:34142', '192.168.xxx.xxx:34184', '192.168.xxx.xxx:55688', '192.168.xxx.xxx:52228', '192.168.xxx.xxx:52200', '192.168.xxx.xxx:57410', '192.168.xxx.xxx:47666', '192.168.xxx.xxx:55394'])}), in 19072, out 8191
2018-12-29 09:51:25,432 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:26,011 - DEBUG - job_exporter.py:107 - pid 22981 with cmd `/bin/bash /usr/local/start.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae410&gt;, {'2213': set(['192.168.xxx.xxx:22']), '18307': set(['192.168.xxx.xxx:22']), '28236': set(['192.168.xxx.xxx:55772', '192.168.xxx.xxx:57476', '192.168.xxx.xxx:34228', '192.168.xxx.xxx:52228', '192.168.xxx.xxx:52200', '192.168.xxx.xxx:35184', '192.168.xxx.xxx:34278', '192.168.xxx.xxx:34280', '192.168.xxx.xxx:47666', '192.168.xxx.xxx:55394', '192.168.xxx.xxx:34266', '192.168.xxx.xxx:34274']), '6012': set(['192.168.xxx.xxx:48680', '192.168.xxx.xxx:52376', '192.168.xxx.xxx:57500', '192.168.xxx.xxx:55758', '192.168.xxx.xxx:55754', '192.168.xxx.xxx:34268', '192.168.xxx.xxx:47790', '192.168.xxx.xxx:35210', '192.168.xxx.xxx:34176']), '23877': set(['192.168.xxx.xxx:50128']), '17978': set(['192.168.xxx.xxx:22']), '33090': set(['192.168.xxx.xxx:26134']), '14951': set(['192.168.xxx.xxx:59522', '192.168.xxx.xxx:5010']), '2129': set(['192.168.xxx.xxx:22']), '29758': set(['192.168.xxx.xxx:32882', '192.168.xxx.xxx:36648']), '20069': set(['192.168.xxx.xxx:30389']), '48563': set(['192.168.xxx.xxx:30389']), '24940': set(['192.168.xxx.xxx:41364', '192.168.xxx.xxx:40126', '192.168.xxx.xxx:36360', '192.168.xxx.xxx:53600']), '20637': set(['192.168.xxx.xxx:20052']), '22889': set(['192.168.xxx.xxx:9100']), '22215': set(['192.168.xxx.xxx:46892', '192.168.xxx.xxx:46896', '192.168.xxx.xxx:46894', '192.168.xxx.xxx:47104', '192.168.xxx.xxx:46898', '192.168.xxx.xxx:46900', '192.168.xxx.xxx:10250', '192.168.xxx.xxx:46928']), '4596': set(['192.168.xxx.xxx:13955']), '19788': set(['192.168.xxx.xxx:50916', '192.168.xxx.xxx:47220']), '15501': set(['192.168.xxx.xxx:25214']), '806': set(['192.168.xxx.xxx:47756', '192.168.xxx.xxx:52370', '192.168.xxx.xxx:55806', '192.168.xxx.xxx:55810', '192.168.xxx.xxx:51488', '192.168.xxx.xxx:57496', '192.168.xxx.xxx:35202', '192.168.xxx.xxx:55622', '192.168.xxx.xxx:55626', '192.168.xxx.xxx:34252']), '22424': set(['192.168.xxx.xxx:46902', '192.168.xxx.xxx:46910'])}), in 212328, out 22299421
2018-12-29 09:51:26,011 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:26,188 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:26,287 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:26,311 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:26,755 - DEBUG - job_exporter.py:107 - pid 5160 with cmd `/dev/init -- /bin/bash /pai/bootstrap/docker_bootstrap.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae5f0&gt;, {'281': set(['192.168.xxx.xxx:20052']), '237': set(['192.168.xxx.xxx:48680', '192.168.xxx.xxx:52452', '192.168.xxx.xxx:57572', '192.168.xxx.xxx:34268', '192.168.xxx.xxx:35222', '192.168.xxx.xxx:47872', '192.168.xxx.xxx:34176', '192.168.xxx.xxx:55844'])}), in 15386, out 6565
2018-12-29 09:51:26,756 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:26,835 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:27,239 - DEBUG - job_exporter.py:107 - pid 33022 with cmd `/dev/init -- /bin/bash /pai/bootstrap/docker_bootstrap.sh
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae410&gt;, {'262': set(['192.168.xxx.xxx:30389']), '403': set(['192.168.xxx.xxx:30389'])}), in 0, out 0
2018-12-29 09:51:27,239 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:27,295 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:30,212 - DEBUG - job_exporter.py:107 - pid 23119 with cmd `/bin/sh -c python /job_exporter/job_exporter.py /datastorage/prometheus 30
` has lsof result defaultdict(&lt;function &lt;lambda&gt; at 0x7f18218ae5f0&gt;, {'2213': set(['192.168.xxx.xxx:22']), '18307': set(['192.168.xxx.xxx:22']), '28236': set(['192.168.xxx.xxx:34682', '192.168.xxx.xxx:52848', '192.168.xxx.xxx:48148', '192.168.xxx.xxx:34686', '192.168.xxx.xxx:52832', '192.168.xxx.xxx:56178', '192.168.xxx.xxx:35554', '192.168.xxx.xxx:57938', '192.168.xxx.xxx:55394']), '6012': set(['192.168.xxx.xxx:48680', '192.168.xxx.xxx:57870', '192.168.xxx.xxx:56248', '192.168.xxx.xxx:35652', '192.168.xxx.xxx:48252', '192.168.xxx.xxx:34736', '192.168.xxx.xxx:52844']), '23877': set(['192.168.xxx.xxx:50128']), '17978': set(['192.168.xxx.xxx:22']), '33090': set(['192.168.xxx.xxx:26134']), '14951': set(['192.168.xxx.xxx:59522', '192.168.xxx.xxx:5010']), '2129': set(['192.168.xxx.xxx:22']), '29758': set(['192.168.xxx.xxx:32882', '192.168.xxx.xxx:36648']), '20069': set(['192.168.xxx.xxx:30389']), '48563': set(['192.168.xxx.xxx:30389']), '24940': set(['192.168.xxx.xxx:40126', '192.168.xxx.xxx:36360']), '20637': set(['192.168.xxx.xxx:20052']), '22889': set(['192.168.xxx.xxx:9100']), '22215': set(['192.168.xxx.xxx:46892', '192.168.xxx.xxx:46896', '192.168.xxx.xxx:46894', '192.168.xxx.xxx:47104', '192.168.xxx.xxx:46898', '192.168.xxx.xxx:46900', '192.168.xxx.xxx:10250', '192.168.xxx.xxx:46928']), '4596': set(['192.168.xxx.xxx:13955']), '19788': set(['192.168.xxx.xxx:50916', '192.168.xxx.xxx:47220']), '15501': set(['192.168.xxx.xxx:25214']), '806': set(['192.168.xxx.xxx:52862', '192.168.xxx.xxx:34632', '192.168.xxx.xxx:52866', '192.168.xxx.xxx:52864', '192.168.xxx.xxx:35628', '192.168.xxx.xxx:57892', '192.168.xxx.xxx:51488', '192.168.xxx.xxx:48232', '192.168.xxx.xxx:56234', '192.168.xxx.xxx:52784']), '22424': set(['192.168.xxx.xxx:46902', '192.168.xxx.xxx:46910'])}), in 212328, out 22299421
2018-12-29 09:51:30,212 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
2018-12-29 09:51:30,395 - DEBUG - docker_inspect.py:60 - ready to run docker inspect
```
- node exporter
```
time="2018-12-24T01:57:31Z" level=info msg="Starting node_exporter (version=0.16.0, branch=HEAD, revision=d42bd70f4363dced6b77d8fc311ea57b63387e4f)" source="node_exporter.go:82"
time="2018-12-24T01:57:31Z" level=info msg="Build context (go=go1.9.6, user=root@a67a9bc13a69, date=20180515-15:52:42)" source="node_exporter.go:83"
time="2018-12-24T01:57:31Z" level=info msg="Enabled collectors:" source="node_exporter.go:90"
time="2018-12-24T01:57:31Z" level=info msg=" - cpu" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - diskstats" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - filefd" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - filesystem" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - loadavg" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - meminfo" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - netdev" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - netstat" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - textfile" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg=" - uname" source="node_exporter.go:97"
time="2018-12-24T01:57:31Z" level=info msg="Listening on :9100" source="node_exporter.go:111"
time="2018-12-26T19:05:39Z" level=error msg="Error parsing \"/datastorage/prometheus/job_exporter.prom\": text format parsing error in line 55: unexpected end of input stream" source="textfile.go:212"
```

**Comment by xudifsd** [comment]
@ICEORY please remove all DEBUG log from job-exporter, also please check if job-exporter's log hangs in somewhere, that hangs will cause job-exporter stop emitting all metrics. Do you have high GPU usage? from our experiences nvidia-smi is prone to hanging, this hanging will make job-exporter can not check gpu usage.

**Comment by ICEORY** [comment]
Yes, it happens when the usage of GPU is high.

**Comment by xudifsd** [comment]
Ok, in that case I think this is a known issue, and this is due to nvidia-smi not due to pai, pai can do nothing about it.

**Comment by xudifsd** [comment]
@fanyangCS in master branch, the latency of nvidia-smi call has already reported, rest server can query prometheus using `histogram_quantile(0.95, sum(rate(cmd_nvidia_smi_latency_seconds_bucket{instance="$node"}[5m])) by (le))` to get 95th percentile of nvidia-smi call, it will return a float number in second.

**Comment by fanyangCS** [comment]
@xudifsd, do you mean you have fixed this issue in the next release?

**Comment by xudifsd** [comment]
No, I mean if we want to display nvidia-smi hangs in UI, the metrics is already available.

**Comment by scarlett2018** [comment]
@ICEORY - thanks for reporting the issue, in the new 2019 year, we will start a set of community communication program for PAI users. Would you mind to share which organization (EDU or Company) you are presenting)? and what's the best way to reach your team, thanks.

**Comment by ICEORY** [comment]
@scarlett2018 my email: zwzhuang@robosense.cn, thanks

**Comment by scarlett2018** [comment]
@ICEORY - well received, thanks!

**Comment by fanyangCS** [comment]
fixed in 0.10.y. the number now relies on yarn.
